src/extension.ts

```diff
@@ -221,6 +221,25 @@ export async function activate(ctx: vscode.ExtensionContext) {
     }
 
     if (configureOnOpen === true) {
+      if (mesonFiles.length > 1) {
+        const items = mesonFiles.map((file, index) => ({ index: index, label: relative(root, file.fsPath) }));
+        items.sort((a, b) => a.label.localeCompare(b.label));
+        let selection;
+        try {
+          selection = await vscode.window.showQuickPick(items, {
+            canPickMany: false,
+            title: "Select configuration to use.",
+            placeHolder: "path/to/meson.build",
+          });
+        } catch (err) {
+          // Pick cancelled.
+        }
+        if (selection) {
+          if (mesonFiles[selection.index] !== mesonFile) {
+            vscode.commands.executeCommand("workbench.action.reloadWindow");
+          }
+        }
```

Comment:
Should we just move this into the try block? Also, let's just combine the two boolean checks into 1.

Reply:
Yeah, the comment is wrong. The following does not work:

![obraz](https://github.com/mesonbuild/vscode-meson/assets/600573/e27fd923-d8d8-4fad-9d0c-46d6f0ad1237)

Reply:
I have removed the try/catch section and added code do prevent reconfigure running if selection was cancelled.